cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -119,41 +147,16 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 		}
 	}
 
-	var r []error
-	var targetResults []*target.TargetResult
-
-	for _, t := range args.Targets {
-		switch options := t.Options.(type) {
-		case *target.LocalTargetOptions:
-			if !osbuildOutput.Success {
-				continue
-			}
-			var f *os.File
-			imagePath := path.Join(outputDirectory, options.Filename)
-			if options.StreamOptimized {
-				f, err = vmware.OpenAsStreamOptimizedVmdk(imagePath)
-				if err != nil {
-					r = append(r, err)
-					continue
-				}
-			} else {
-				f, err = os.Open(imagePath)
-				if err != nil {
-					r = append(r, err)
-					continue
-				}
-			}
-
-			err = job.UploadArtifact(options.Filename, f)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
+	if len(args.Targets) == 0 {
+		// There is no upload target, mark this job a success.
+		osbuildJobResult.Success = true
+		osbuildJobResult.UploadStatus = "success"
+	} else if len(args.Targets) == 1 {
+		// Although the API takes a list of targets, we only support one upload target
+		// per job, so we only need to handle this case and panic in case of multiple
```

Comment:
The comment talks about "panic", but there is no panic... And probably IMHO should not be, because providing multiple targets is not a programmer's error, but caller's error(?)

Reply:
You are right, I forgot to remove this comment. See the "if" statement checking for a single upload target. It returns a failure to osbuild-composer. I think that makes more sense, WDYT?